meerkat/mixins/groupbyable.py

```diff
@@ -0,0 +1,59 @@
+
+
+
+from abc import ABC, abstractmethod
+
+class BaseGroupBy(ABC):
+    def __init__(self, indices, data, by, keys) -> None:
+        self.indices = indices
+        self.data = data
+        self.by = by
+        self.keys = keys
+
+    def mean(self):
+        # inputs: self.indices are a dictionary of {
+        #   labels : [indices]
+        # }
+        labels = list(self.indices.keys())
+
+        # sorting them so that they appear in a nice order.
+        labels.sort()
+
+
+        print("Mean", labels)
+
+        # Means will be a list of dictionaries where each element in the dict
+
+        means = []
+        for label in labels:
```

Comment:
Let's discuss how we can allow some column types to implement optimized versions of this groupby operation. Probably out of scope for this PR, but we will want to support it in the future